src/jit/morph.cpp

```diff
@@ -600,6 +600,7 @@ GenTree* Compiler::fgMorphCast(GenTree* tree)
                             case GT_LCL_FLD:
                             case GT_ARR_ELEM:
                                 oper->gtType = dstType;
+                                oper->SetVNsFromNode(tree);
```

Comment:
I think this is worth a comment explaining why this isn't done for the default case.

Reply:
Sorry - I meant the other cases where we `goto REMOVE_CAST;` (note the default). Someone reading this might wonder why we don't need to regenerate the VNs in those other cases.

Reply:
VNs need to be regenerated only when oper is changed (e.g., its type is changed). The only other `goto REMOVE_CAST;`place where that happens is where we call `optNarrowTree`. `optNarrowTree `does update the value number using its fourth argument.

Reply:
Right. So a simple "We're changing the type here so we need a new VN; in other cases (constants and implicit loads) the VN doesn't change." would suffice, I think.

Reply:
Thanks, I added a comment.